src/Core/Billing/Utilities.cs

```diff
@@ -8,4 +12,67 @@ public static class Utilities
         string internalMessage = null,
         Exception innerException = null) => new("Something went wrong with your request. Please contact support.",
         internalMessage, innerException);
+
+    public static async Task<SubscriptionSuspensionDTO> GetSuspensionAsync(
```

Comment:
I was going to add this to the `SubscriberService`, but in order to reduce the number of calls to retrieve a `Subscription` from Stripe, I found it easier to just provide a utility method that takes the `StripeAdapter` as a dependency. This way, we can use an already retrieved `Subscription`.